src/services/identity/SitesService.ts

```diff
@@ -154,29 +261,13 @@ class SitesService {
   }
 
   async getStagingUrl(sessionData: UserWithSiteSessionData): Promise<string> {
-    // Check config.yml for staging url if it exists, and github site description otherwise
-    const { content: configData } = await this.configYmlService.read(
-      sessionData
-    )
-    if ("staging" in configData) return configData.staging
-
-    const {
-      description,
-    }: { description: string } = await this.gitHubService.getRepoInfo(
-      sessionData
-    )
-
-    if (description) {
-      // Retrieve the url from the description - repo descriptions have varying formats, so we look for the first link
-      const descTokens = description.replace("/;/g", " ").split(" ")
-      // Staging urls also contain staging in their url
-      const stagingUrl = descTokens.find(
-        (token) => token.includes("http") && token.includes("staging")
+    const { staging } = await this.getUrlsOfSite(sessionData, ["staging"])
+    if (!staging) {
+      throw new NotFoundError(
+        `${sessionData.siteName} does not have a staging url`
```

Comment:
shouldn't throw, prefer returning an error and let the router decide what to do

Reply:
Changed in ed61f55.